src/app/components/DynamicContentStyles.tsx

```diff
@@ -11,25 +17,46 @@ export const WithStyles = styled.div`
 
 const cacheStyles = new Map<string, string>();
 
+const getCssFileUrl = (
+  queryParams: OutputParams, book: State['book'], archiveLoader: AppServices['archiveLoader']
+) => {
+  if (queryParams['content-style']) {
+    return queryParams['content-style'];
+  } else if (book) {
+    const bookConfig = getBookVersionFromUUIDSync(book.id);
+
+    if (bookConfig?.dynamicStyles && book.style_href) {
+      if (isAbsoluteUrl(book.style_href)) {
+        return book.style_href;
+      } else {
+        const contentUrl = archiveLoader.book(book.id, book.version).url();
+        return fromRelativeUrl(contentUrl, book.style_href);
+      }
+    }
+  }
+};
+
 interface DynamicContentStylesProps extends React.HTMLAttributes<HTMLDivElement> {
+  book: State['book'];
   disable?: boolean;
 }
 
 // tslint:disable-next-line: variable-name
 const DynamicContentStyles = React.forwardRef<HTMLElement, DynamicContentStylesProps>((
-  { children, disable, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>,
+  { book, children, disable, ...otherProps }: React.PropsWithChildren<DynamicContentStylesProps>,
   ref
 ) => {
-  const queryParams = useSelector(query);
   const [styles, setStyles] = React.useState('');
+  const queryParams = useSelector(query);
+  const { archiveLoader } = useServices();
+  const cssfileUrl = getCssFileUrl(queryParams, book, archiveLoader);
```

Comment:
you probably want to wrap this in React.useMemo or it could result in the following effect running on every render

Reply:
or move the call inside the useEffect

Reply:
I don't know why I had moved it out of useEffect... anyway I put it back in.